crypto/bio/bss_mem.c

```diff
@@ -299,7 +299,7 @@ static long mem_ctrl(BIO *b, int cmd, long num, void *ptr)
         ret = (long)bm->length;
         if (ptr != NULL) {
             pptr = (char **)ptr;
-            *pptr = (char *)&(bm->data[0]);
+            *pptr = (char *)(bm->data);
```

Comment:
Does this actually emit different code?

Reply:
ubsan was complaining about the addition of 0 to a NULL pointer. So if `bm->data` is NULL then `&(bm->data[0])` is undefined!!

Reply:
Interesting. I thought most compilers optimizer away the "difference". Ok then

Reply:
Well, yeah. I guess they do. But ubsan's job is to be picky about such things.

Reply:
The compiler is going to assume it has no undefined behaviour and compile both to the same thing. The reason to change the code is to make ubsan not give an error for the undefined behaviour.

Reply:
Oh of course. I was just a bit confused as to when that sanitizer operates...